### ISSUE TITLE
Problem with loader `is_package` prevents binary from finding files

### ISSUE THREAD

**Issue by anekix** [Nuitka/Nuitka#461]
## Error on running binary generated by Nuitka
`SystemError: More keyword list entries (2) than format specifiers (1)
`

#### Nutika version:
```
0.6.4
Python: 3.7.3 (default, May 11 2019, 00:45:16)
Executable: /usr/bin/python3
OS: Linux
Arch: x86_64
```

#### installation method
`Pip  `

#### command used
`python3 -m nuitka --standalone --follow-imports --recurse-all --show-progress --show-scons run.py`

`run.py` is the entry point of flask application

I am succesfully able to build my flask application  but when i attempt to run it i get these errors:

```
Traceback (most recent call last):
  File "/home/test/app/run.dist/run.py", line 36, in <module>
  File "/home/test/app/run.dist/main/__init__.py", line 14, in create_app
  File "/home/test/run.dist/flask/app.py", line 380, in __init__
  File "/home/test/app/run.dist/flask/app.py", line 677, in auto_find_instance_path
  File "/home/test/app/run.dist/flask/helpers.py", line 818, in find_package
  File "/home/test/app/run.dist/flask/helpers.py", line 769, in _matching_loader_thinks_module_is_package
SystemError: More keyword list entries (2) than format specifiers (1)
```
i ran the binary with `strace`  and i found the log below which means somehow some required files are not found .what could be the reason , is there any specific flag that i missed which i have to include while compiling?

```
write(2, "  File \"/home/test/"..., 94  File "/home/test/app/run.dist/run.py", line 36, in <module>
) = 94
openat(AT_FDCWD, "/home/test/app/run.dist/run.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/run.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/run.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "  File \"/home/test/"..., 106  File "/home/test/app/run.dist/main/__init__.py", line 14, in create_app
) = 106
openat(AT_FDCWD, "/home/test/app/run.dist/main/__init__.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/__init__.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/__init__.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "  File \"/home/test/"..., 101  File "/home/test/app/run.dist/flask/app.py", line 380, in __init__
) = 101
openat(AT_FDCWD, "/home/test/app/run.dist/flask/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "  File \"/home/test/"..., 116  File "/home/test/app/run.dist/flask/app.py", line 677, in auto_find_instance_path
) = 116
openat(AT_FDCWD, "/home/test/app/run.dist/flask/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/app.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "  File \"/home/test/"..., 109  File "/home/test/app/run.dist/flask/helpers.py", line 818, in find_package
) = 109
openat(AT_FDCWD, "/home/test/app/run.dist/flask/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "  File \"/home/test/"..., 138  File "/home/test/app/run.dist/flask/helpers.py", line 769, in _matching_loader_thinks_module_is_package
) = 138
openat(AT_FDCWD, "/home/test/app/run.dist/flask/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/test/app/run.dist/helpers.py", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "SystemError: More keyword list e"..., 70SystemError: More keyword list entries (2) than format specifiers (1)
```
app directory structure inside **_home/test/app_** is :

```
├── main
│   ├── admin
│   │   ├── applications
│   │   └── static
│   ├── commons
│   ├── errors
│   ├── pki
│   ├── public
│   ├── templates
│   └── users
├── run.build
├── run.dist
│   ├── bson
│   ├── certifi
│   ├── Crypto
│   │   ├── Cipher
│   │   ├── Hash
│   │   ├── Protocol
│   │   └── Util
│   ├── cryptography
│   │   └── hazmat
│   │       └── bindings
│   ├── gi
│   ├── markupsafe
│   ├── problem
│   ├── pymongo
│   ├── report
│   └── systemd
└── tests

```

**Comment by kayhayen** [comment]
So, your strace output checks for Python files, and that is because creating a traceback tries to look up the source code, and that is not contained in the dist folder, but it looks there. No harm in that. Tracebacks just won't contain source code lines. Something I want to make optional in the future btw, but many standalone users do not care about that. Contrary, they want no source code to be included.

However, can you post the code of `_matching_loader_thinks_module_is_package` here, the SystemError seems weird. Does it raise that itself?

**Comment by anekix** [comment]
@kayhayen This is part of flask framework. i am pasting and linking the source for [_matching_loader_thinks_module_is_package](https://github.com/pallets/flask/blob/722d5e610a1f11ef2010dd719811771706ea0683/src/flask/helpers.py#L822 )

```
def _matching_loader_thinks_module_is_package(loader, mod_name):
    """Given the loader that loaded a module and the module this function
    attempts to figure out if the given module is actually a package.
    """
    # If the loader can tell us if something is a package, we can
    # directly ask the loader.
    if hasattr(loader, "is_package"):
        return loader.is_package(mod_name)
    # importlib's namespace loaders do not have this functionality but
    # all the modules it loads are packages, so we can take advantage of
    # this information.
    elif (
        loader.__class__.__module__ == "_frozen_importlib"
        and loader.__class__.__name__ == "NamespaceLoader"
    ):
        return True
    # Otherwise we need to fail with an error that explains what went
    # wrong.
    raise AttributeError(
        (
            "%s.is_package() method is missing but is required by Flask of "
            "PEP 302 import hooks.  If you do not use import hooks and "
            "you encounter this error please file a bug against Flask."
        )
        % loader.__class__.__name__
    )

```

**Comment by kayhayen** [comment]
It looks for meas if the format string was making the complaint, which it may well do either day, but from the looks of it, should never do. I have a suspect this means something is wrong about this:

```
static PyObject *_path_unfreezer_is_package(PyObject *self, PyObject *args, PyObject *kwds) {
    PyObject *module_name;

    int res = PyArg_ParseTupleAndKeywords(args, kwds, "O:is_package", _kwlist, &module_name);

    if (unlikely(res == 0)) {
        return NULL;
    }

    assert(module_name);
    assert(Nuitka_String_Check(module_name));

    char const *name = Nuitka_String_AsString(module_name);

    struct Nuitka_MetaPathBasedLoaderEntry *entry = findEntry(name);

    PyObject *result;

    if (entry) {
        result = BOOL_FROM((entry->flags & NUITKA_PACKAGE_FLAG) != 0);
    } else {
        // TODO: Maybe needs to be an exception.
        result = Py_None;
    }

    Py_INCREF(result);
    return result;
}

```

Oh yeah, checking `kw_list`, that doesn't match the signature, has 2 arguments defined. But there will only be one (this being a static method, no self). I will release something to address this shortly.

It ought to be a regression of my attempts to make the loader objects and obviously fixable.

Yours,
Kay

**Comment by anekix** [comment]
@kayhayen Thanks. we are planning to use Nuitka as the main compiler for our project. if i can help in any way to work on this fix let me know.

**Comment by kayhayen** [comment]
This is part of the latest pre-release and should work now. Will be part of 0.6.5

**Comment by anekix** [comment]
@kayhayen Thanks for working on this issue .how can i get the fixed version . is it in some specific branch?

**Comment by kayhayen** [comment]
The tag develop indicates that.

**Comment by kayhayen** [comment]
Released as 0.6.5
